src/components/users-list.tsx

```diff
@@ -1,18 +1,37 @@
 import {useQuery} from '@apollo/client';
-import React from 'react';
-import {
-  ActivityIndicator,
-  FlatList,
-  ListRenderItemInfo,
-  SafeAreaView,
-  Text,
-} from 'react-native';
+import React, {useState} from 'react';
+import {ActivityIndicator, FlatList, SafeAreaView, Text} from 'react-native';
 import {usersQuery} from '../utils/apollo';
 import {UserNode} from '../utils/models';
-import {UserItem} from './user-item';
+import {Loading, renderItem} from './list-functions';
+
+const limit = 15;
 
 export const UsersList = () => {
-  const {data, loading} = useQuery(usersQuery);
+  const [offset, setOffset] = useState(0);
+  const [hasMoreData, setHasMoreData] = useState(true);
+  const [list, setList] = useState<UserNode[]>([]);
+  const {refetch, loading} = useQuery(usersQuery, {
+    variables: {data: {limit: limit, offset: 0}},
+    onCompleted: data => {
+      setList(prev => [...prev, ...data.users.nodes]);
+      setHasMoreData(data.users.pageInfo.hasNextPage);
```

Comment:
Esse é um detalhe que quase não faz diferença prática, mas acho que vale pra aprendizado: essa informação se tem ou não próxima página não é utilizada para renderizar algo ou para algume stado diferente da tela, certo? É apenas uma informação que vc quer consultar em determinado momento do lifecycle. Então vc poderia usar ref em vez de estado. Assim quando vc precisa trocar esse valor, você não trigga um render adicional desnecessário.

Reply:
isso vale para o outro estado também em, do offset. Inclusive, aqui n seria necessário nem utilizar estado/ref, daria pra usar o resultado da query diretamente, não?

Reply:
Só cuidado que se vc for seguir a sugestão do primeiro comment, o offset vai precisar ser um estado sim 🤣